Engine/Server/Src/GameEngine/Room/Room.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type Server
+** File description:
+** Room
+*/
+
+#include "Room.hpp"
+
+GameEngine::Room::Room(ABINetwork::roomInfo_t roomInfo)
+{
+    if (roomInfo.name.empty())
+        throw RoomError("A room must has a name");
+    if (roomInfo.isPrivate && roomInfo.password.empty())
+        throw RoomError("A private room must has a password");
+    _roomInfos = roomInfo;
+
+    _roomServer = ABINetwork::createServer(_roomInfos.playerMax);
+    _registries = std::make_shared<ECS::Registry>();
+    _sceneManager = std::make_shared<SceneManager::ServerSceneManager>(_registries);
```

Comment:
Are you up to main ? Because the ASceneManager constructor changed, so the its children too, now the scene manager takes a EventListener as parameter too

Reply:
I'm not totally up to date with main, this is the goal of the next issue.
Link the main/all buttons with the new ABI.

Reply:
Ok perfect, I just forgot that you were not merging into main, mb